Google.Api.Generator/CodeGenerator.cs

```diff
@@ -169,6 +177,29 @@ internal static class CodeGenerator
             }
 
             ValidateTransports(transports, allServiceDetails);
+            return resultFiles;
+
+            // Returns the result files we should actually create, ignoring mix-ins unless we're actually
+            // generating the mix-in API.
+            // This is necessary because Bazel passes more files to generate than we really want.
+            // Note that this method may modify filesByProtoPackage - we're not using it after calling this method though.
+            static List<ResultFile> GetResultFilesToCreate(Dictionary<string, List<ResultFile>> filesByProtoPackage)
+            {
+                var mixinPackages = AllowedAdditionalServices.Select(svc => svc.File.Package).ToHashSet();
+
+                // Unless we're *only* generating mixins, remove all mixins.
```

Comment:
I guess mixins don't have a Common package now, but they might in the future or something. Maybe just pass to this method the name of the main service, which is already stored on the caller I think.

Reply:
Well, we guess at the name of the main service by finding the first generated service... which is no more definitive than this, really. I very, very much doubt that mixins will start having other dependencies. We can cross that bridge when we come to it.